### PR TITLE
Dump store to logs in circle

### PR DESCRIPTION
**Pull request by andrewgordstewart**
Aiming to reproduce https://github.com/statechannels/monorepo/issues/1478, I wrote code to dump the store on command.

While I attempt to [reproduce the issue on circle](https://github.com/statechannels/monorepo/commit/7d109642599dd77f8aeab3a946ff53ea7f4dc5a2) (I have been unable to locally) I think this code will be useful in `master`.

```
❯ cat seed-download.e2e.log | grep '"store":' | jq .store.ledgers
{}
{
  "firebase:simple-hub": "0x3d59eb841d26f1e7c5be21c1043bc3902b6b991581c22982772f2511c419b274"
}
{}
{
  "firebase:simple-hub": "0x71ea10595d1bbee25a7cb4c3b2cdf89752cb9ad0e47a86df22050e617256d32c"
}
```

[This artifact](https://19680-209829093-gh.circle-artifacts.com/0/tmp/seed-download.pino.log) is still under 1MB, so it does not seem to generate very large logs

### CODE REVIEW

File: packages/client-api-schema/src/generated-schema.json [hunk 1]
```diff
@@ -705,13 +705,6 @@
       ],
       "type": "object"
     },
-    "ErrorCode": {
-      "enum": [
-        200,
-        100
-      ],
-      "type": "number"
-    },
```
Comment (snario) [inline packages/client-api-schema/src/generated-schema.json:714]: Why is this here?

File: packages/xstate-wallet/src/channel-wallet.ts [hunk 2]
```diff
@@ -146,13 +145,23 @@ export class ChannelWallet {
...
+    const service = interpret(machine, {devTools})
       .onTransition((state, event) => ADD_LOGS && logTransition(state, event, workflowId))
+      .onTransition(
+        async () =>
+          IS_PRODUCTION &&
+          logger.info({workflowId, store: await this.store.dumpBackend()}, 'Done workflow')
+      )
       .onDone(() => (this.workflows = this.workflows.filter(w => w.id !== workflowId)))
+      .onStop(
+        async () =>
```
Comment (snario) [inline packages/xstate-wallet/src/channel-wallet.ts:161]: Does this need to be `async`?

File: packages/xstate-wallet/src/store/indexedDB-backend.ts [hunk 3]
```diff
@@ -17,7 +17,7 @@ enum ObjectStores {
 
 // A running, functioning example can be seen and played with here: https://codesandbox.io/s/elastic-kare-m1jp8
 export class IndexedDBBackend implements DBBackend {
-  private _db: any;
+  private _db: IDBDatabase;
```
Comment (snario) [inline packages/xstate-wallet/src/store/indexedDB-backend.ts:20]: Where is this type defined?

File: packages/xstate-wallet/src/store/indexedDB-backend.ts [hunk 4]
```diff
@@ -61,11 +62,40 @@ export class IndexedDBBackend implements DBBackend {
       request.onerror = err => reject(err);
       request.onsuccess = () => {
         this._db = request.result;
+
         resolve(request.result);
       };
     });
   }
 
+  public async dump(): Promise<object> {
```
Comment (snario) [inline packages/xstate-wallet/src/store/indexedDB-backend.ts:71]: Is `dump` the right name for this? Maybe `serialize` or `toJSON` or something?

File: packages/xstate-wallet/src/workflows/application.ts [hunk 5]
```diff
@@ -121,6 +122,7 @@ const generateConfig = (
   actions: WorkflowActions,
   guards: WorkflowGuards
 ): MachineConfig<WorkflowContext, WorkflowStateSchema, WorkflowEvent> => ({
+  strict: !IS_PRODUCTION,
```
Comment (snario) [inline packages/xstate-wallet/src/workflows/application.ts:125]: Why?

File: packages/client-api-schema/src/generated-schema.json (hunk 1 above)
Comment (andrewgordstewart) [reply packages/client-api-schema/src/generated-schema.json:714]: I needed to run `yarn prepare` to update something (I think?), and it resulted in this change: https://github.com/statechannels/monorepo/pull/1491/commits/e47d753dae92d898c1a4585c215cdb4c3bcaec3a

File: packages/xstate-wallet/src/channel-wallet.ts (hunk 2 above)
Comment (andrewgordstewart) [reply packages/xstate-wallet/src/channel-wallet.ts:161]: Not anymore!
```suggestion
        () =>
```

File: packages/xstate-wallet/src/store/indexedDB-backend.ts (hunk 3 above)
Comment (andrewgordstewart) [reply packages/xstate-wallet/src/store/indexedDB-backend.ts:20]: I would guess in the indexedDB type package?

File: packages/xstate-wallet/src/workflows/application.ts (hunk 5 above)
Comment (andrewgordstewart) [reply packages/xstate-wallet/src/workflows/application.ts:125]: I was attempting to be able to log the store dump only on error: https://github.com/statechannels/monorepo/pull/1491/commits/8b7d3df4b01072235eb7b918e1ddcbf57de20ed1

Eventually, I decided to log the store on every transition, because it doesn't not really pollute the logs.

Perhaps it should just be set to `strict: true`, but I am not sure if we want that.

File: packages/xstate-wallet/src/store/indexedDB-backend.ts (hunk 4 above)
Comment (andrewgordstewart) [reply packages/xstate-wallet/src/store/indexedDB-backend.ts:71]: Perhaps, but it is not something I put a lot of thought into.

The purpose of this is to log the store contents for debugging purposes. So, `dump` may seem more appropriate than `serialize`, considering its usage in programming.

I'm not fussed either way.

**Comment by andrewgordstewart** [comment]
Closed, as the indexedDB backend is now removed.
